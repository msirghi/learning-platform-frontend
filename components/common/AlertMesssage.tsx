import React from 'react';
import Alert from '@material-ui/lab/Alert';
import { AlertType } from '../../common/enums';

/**
 * AlertMessage component props.
 */
type Props = {
  /**
   * Message that will be displayed
   */
  message: string;
  /**
   * Alert type. Depending on it component will have special colors.
   */
  type: AlertType;
};

/**
 * Component for displaying the alert messages. Uses Alert component from '@material-ui/lab' library.
 *
 * @version 0.1
 * @author [Sirghi Mihail](https://github.com/msirghi)
 */
export const AlertMessage: React.FC<Props> = ({ message, type }) => {
  return <Alert severity={type}>{message}</Alert>;
};
