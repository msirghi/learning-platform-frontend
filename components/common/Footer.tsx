import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';

function Copyright() {
  return (
    <Typography variant='body2'>
      {'Copyright Â© '}
      <Link color='inherit' href='https://material-ui.com/'>
        Your Website
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor: '#202020',
    color: '#fff'
  }
}));

export const SiteFooter = () => {
  const classes = useStyles();
  return (
    <footer className={classes.footer}>
      <Container maxWidth='sm'>
        <Typography variant='body1'>My sticky footer can be found here.</Typography>
        <Copyright />
      </Container>
    </footer>
  );
};
