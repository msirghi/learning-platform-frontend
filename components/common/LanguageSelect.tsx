import React, { useState } from 'react';
import LanguageIcon from '@material-ui/icons/Language';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Menu from '@material-ui/core/Menu';
import { useTranslation } from '../../i18n';
import { useDispatch } from 'react-redux';
import { setInterfaceLocale } from '../../redux/actions/preference/preferenceAction';

const languageChoice = [
  {
    key: 'en',
    value: 'English'
  },
  {
    key: 'ru',
    value: 'Русский'
  }
];

export const LanguageSelect: React.FC = () => {
  const [notificationMenuOpen, setNotificationMenuOpen] = useState<null | HTMLElement>(null);
  const { i18n } = useTranslation();
  const dispatch = useDispatch();

  const handleChange = (val: string) => {
    dispatch(setInterfaceLocale(val));
    i18n.changeLanguage(val);
    handleClose();
  };

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setNotificationMenuOpen(event.currentTarget);
  };

  const handleClose = () => {
    setNotificationMenuOpen(null);
  };

  return (
    <>
      <IconButton onClick={handleClick}>
        <LanguageIcon color='primary' />
      </IconButton>
      <Menu
        anchorEl={notificationMenuOpen}
        keepMounted
        open={Boolean(notificationMenuOpen)}
        onClose={handleClose}
      >
        <List>
          {languageChoice.map((item) => {
            return (
              <ListItem key={item.key} button onClick={() => handleChange(item.key)}>
                {item.value}
              </ListItem>
            );
          })}
        </List>
      </Menu>
    </>
  );
};
