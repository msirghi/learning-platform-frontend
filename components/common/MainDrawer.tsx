import React, { useEffect, useState } from "react";
import Drawer from "@material-ui/core/Drawer";
import IconButton from "@material-ui/core/IconButton";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles } from "@material-ui/core/styles";
import clsx from "clsx";
import { menuItems } from "./utils/menuItems";
import styles from "../../styles/modules/Shared.module.scss";
import { Hidden } from "@material-ui/core";

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  drawer: {
    width: drawerWidth,
  },
  drawerOpen: {
    background: theme.palette.primary.main,
    width: drawerWidth,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerClose: {
    background: theme.palette.primary.main,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: "hidden",
    width: theme.spacing(8) + 1,
    [theme.breakpoints.up("sm")]: {
      width: 110,
    },
  },
  toolbar: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
  },
  drawerPaper: {
    width: drawerWidth,
    background: theme.palette.primary.main,
  },
}));

type Props = {
  handleDrawerClose: () => void;
  open: boolean;
  mobileOpen: boolean;
};

export const MainDrawer: React.FC<Props> = ({ handleDrawerClose, open, mobileOpen }) => {
  const classes = useStyles();
  const [activeTab, setActiveTab] = useState<string | null>(null);

  useEffect(() => {
    /* istanbul ignore next */
    if (window) {
      setActiveTab(window.location.href);
    }
  }, []);

  if (!activeTab) {
    return <div />;
  }

  const drawerContent = () => {
    return (
      <>
        <div className={classes.toolbar}>
          <IconButton onClick={handleDrawerClose}>
            <img src='/images/register-logo-148.png' height={72} width={72} />
          </IconButton>
        </div>
        <List className={styles.mainMenuListContainer}>
          {menuItems.map((item) => {
            const isActiveTab = activeTab.includes(item.link);
            return (
              <div
                className={`${styles.mainMenuColor} ${isActiveTab && styles.mainMenuActive} ${
                  item.className
                }`}
                key={item.label}
              >
                <ListItem button>
                  <ListItemIcon className={styles.mainMenuIconContainer}>
                    <item.icon
                      className={`${isActiveTab ? styles.mainMenuActiveIcon : styles.mainMenuIcon}`}
                    />
                  </ListItemIcon>
                  {(open || mobileOpen) && (
                    <ListItemText
                      className={isActiveTab ? styles.mainMenuActiveText : ""}
                      primary={item.label}
                    />
                  )}
                </ListItem>
              </div>
            );
          })}
        </List>
      </>
    );
  };

  return (
    <>
      <Drawer
        variant='temporary'
        open={mobileOpen}
        onClose={handleDrawerClose}
        classes={{
          paper: classes.drawerPaper,
        }}
        ModalProps={{
          keepMounted: true,
        }}
      >
        {drawerContent()}
      </Drawer>
      <Hidden xsDown implementation='css'>
        <Drawer
          classes={{
            paper: clsx({
              [classes.drawerOpen]: open,
              [classes.drawerClose]: !open,
            }),
          }}
          className={clsx(classes.drawer, {
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open,
          })}
          variant='permanent'
          open={open}
        >
          {drawerContent()}
        </Drawer>
      </Hidden>
    </>
  );
};
