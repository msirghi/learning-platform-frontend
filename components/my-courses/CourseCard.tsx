import React from 'react';
import styles from '../../styles/modules/MyCourses.module.scss';
import Button from '@material-ui/core/Button';
import { Course } from '../../common/types';
import { COURSE_DEFAULT_COVER_IMAGE } from '../../common/constants/common.constants';

type Props = {
  course: Course;
};

export const CourseCard: React.FC<Props> = ({
  course: { id, teacherImg, coverImg, name, description }
}) => {
  return (
    <div className={styles.cardWrapper}>
      <div className={styles.cardImage}>
        <img src={coverImg || COURSE_DEFAULT_COVER_IMAGE} alt={description} />
      </div>
      <div className={styles.cardTeacherPhoto}>
        <img src={teacherImg} alt={name} />
      </div>
      <div className={styles.cardContent}>
        <div className={styles.cardContentTitle}>{name}</div>
        <div className={styles.cardContentDescription}>{description}</div>
      </div>
      <div className={styles.cardButtonContainer}>
        <Button variant='outlined' color='primary'>
          Enter
        </Button>
      </div>
    </div>
  );
};
