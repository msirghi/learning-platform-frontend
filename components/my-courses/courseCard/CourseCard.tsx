import React from 'react';
import Button from '@material-ui/core/Button';
import { useRouter } from 'next/router';
import { Course } from '../../../common/types';
import { useTranslation } from '../../../i18n';
import { COURSE_DEFAULT_COVER_IMAGE } from '../../../common/constants/common.constants';
import styles from './CourseCard.module.scss';

type Props = {
  course: Course;
};

const CourseCard: React.FC<Props> = ({
  course: { id, teacherImg, coverImg, name, description }
}) => {
  const router = useRouter();

  const { t } = useTranslation();
  return (
    <div className={styles.cardWrapper}>
      <div className={styles.cardImage}>
        <img src={coverImg || COURSE_DEFAULT_COVER_IMAGE} alt={description} />
      </div>
      <div className={styles.cardTeacherPhoto}>
        <img src={teacherImg} alt={name} />
      </div>
      <div className={styles.cardContent}>
        <div className={styles.cardContentTitle}>{name}</div>
        <div className={styles.cardContentDescription}>{description}</div>
      </div>
      <div className={styles.cardButtonContainer}>
        <Button variant='outlined' color='primary' onClick={() => router.push(`/myCourses/${id}`)}>
          {t('courses:enterButtonLabel')}
        </Button>
      </div>
    </div>
  );
};

export default CourseCard;
