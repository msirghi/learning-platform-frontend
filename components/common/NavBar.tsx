import React, { useEffect, useState } from 'react';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import Typography from '@material-ui/core/Typography';
import NotificationsIcon from '@material-ui/icons/Notifications';
import styles from '../../styles/modules/Shared.module.scss';
import Divider from '@material-ui/core/Divider';
import { UserAvatar } from './UserAvatar';
import Badge from '@material-ui/core/Badge';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import { LanguageSelect } from './LanguageSelect';
import { useTranslation } from '../../i18n';

type Props = {
  classes: { [key: string]: string };
  handleDrawerOpen: () => void;
  handleDrawerClose: () => void;
  open: boolean;
};

export const NavBar: React.FC<Props> = ({ classes, handleDrawerOpen, open, handleDrawerClose }) => {
  const [activeTab, setActiveTab] = useState<string | undefined>();
  const { t, i18n } = useTranslation();

  useEffect(() => {
    if (global.window) {
      console.log('global.window.location :>> ', global.window.location.pathname);
      const href = global.window.location.pathname.replaceAll('/', '');
      setActiveTab(t(`common:headerTabs.${href}`));
    }
  }, [global.window, i18n]);

  return (
    <AppBar
      position='fixed'
      className={clsx(classes.appBar, {
        [classes.appBarShift]: open
      })}
    >
      <Toolbar>
        <IconButton
          color='inherit'
          aria-label='open drawer'
          onClick={handleDrawerOpen}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: open
          })}
        >
          <MenuIcon />
        </IconButton>

        <IconButton
          color='inherit'
          aria-label='close drawer'
          onClick={handleDrawerClose}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: !open
          })}
        >
          <ChevronLeftIcon />
        </IconButton>
        <Typography variant='h6' noWrap>
          {activeTab}
        </Typography>
        <div className={styles.navBarRightIconsContainer}>
          <LanguageSelect />
          <IconButton>
            <Badge badgeContent={4} color='primary'>
              <NotificationsIcon color={'primary'} />
            </Badge>
          </IconButton>
          <IconButton>
            <UserAvatar />
          </IconButton>
        </div>
      </Toolbar>
      <div className={styles.navBarDividerContainer}>
        <Divider light />
      </div>
    </AppBar>
  );
};
