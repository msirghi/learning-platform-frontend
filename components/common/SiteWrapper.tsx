import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { CssBaseline } from '@material-ui/core';
import { NavBar } from './NavBar';
import { MainDrawer } from './MainDrawer';
import { SiteFooter } from './Footer';
import { useWindowSize } from './hooks/useWindowResize';

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh'
  },
  root: {
    display: 'flex'
  },
  main: {
    marginTop: theme.spacing(8),
    marginBottom: theme.spacing(2)
  },
  appBar: {
    backgroundColor: '#fafafa',
    color: '#000',
    boxShadow: 'none',
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth / 2}px)`
    }
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  menuButton: {
    marginRight: 36
  },
  hide: {
    display: 'none'
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3)
  }
}));

export const SiteWrapper: React.FC = ({ children }) => {
  const classes = useStyles();
  const [desktopOpen, setDesktopOpen] = React.useState(false);
  const [mobileOpen, setMobileOpen] = React.useState(false);
  const [width] = useWindowSize();

  useEffect(() => {
    if (width < 768 && desktopOpen) {
      setDesktopOpen(false);
      return;
    }

    if(width > 768 && mobileOpen) {
      setMobileOpen(false);
    }
  }, [width]);

  const handleDrawerOpen = () => {
    if (width > 768) {
      setDesktopOpen(true);
      return;
    }
    setMobileOpen(true);
  };

  const handleDrawerClose = () => {
    if (width > 768) {
      setDesktopOpen(false);
      return;
    }
    setMobileOpen(false);
  };

  return (
    <div className={classes.wrapper}>
      <div className={classes.root}>
        <CssBaseline />
        <NavBar
          classes={classes}
          handleDrawerClose={handleDrawerClose}
          handleDrawerOpen={handleDrawerOpen}
          open={desktopOpen}
        />
        <MainDrawer
          handleDrawerOpen={handleDrawerOpen}
          mobileOpen={mobileOpen}
          handleDrawerClose={handleDrawerClose}
          open={desktopOpen}
        />
        <main className={classes.content}>
          <div className={classes.toolbar} />
          {children}
        </main>
      </div>
      <SiteFooter />
    </div>
  );
};
