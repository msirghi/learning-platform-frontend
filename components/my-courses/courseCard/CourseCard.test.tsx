import { shallow } from 'enzyme';
import React from 'react';
import CourseCard from './CourseCard';
import { coursesMock } from '../../../mocks/coursesMock';
import toJson from 'enzyme-to-json';
import { COURSE_DEFAULT_COVER_IMAGE } from '../../../common/constants/common.constants';
import { Button } from '@material-ui/core';

describe('CourseCard component', () => {
  /* eslint-disable */
  const useRouter = jest.spyOn(require('next/router'), 'useRouter');
  const pushSpy = jest.fn();

  useRouter.mockImplementation(() => ({
    route: '/yourRoute',
    pathname: '/yourRoute',
    query: '',
    asPath: '',
    push: pushSpy
  }));

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<CourseCard course={coursesMock[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have default image cover if there is no cover image', () => {
    const wrapper = shallow(<CourseCard course={{ ...coursesMock[0], coverImg: null }} />);
    const img = wrapper.find('img').at(0);
    expect(img.props().src).toEqual(COURSE_DEFAULT_COVER_IMAGE);
  });

  it('should redirect on button click', () => {
    const wrapper = shallow(<CourseCard course={{ ...coursesMock[0], coverImg: null }} />);
    const button = wrapper.find(Button);

    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(pushSpy).toBeCalled();
  });
});
