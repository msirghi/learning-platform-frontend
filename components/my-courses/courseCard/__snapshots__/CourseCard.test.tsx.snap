// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard component should match the snapshot 1`] = `
<div
  className="cardWrapper"
>
  <div
    className="cardImage"
  >
    <img
      alt="Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero non illum est fuga beatae ab, assumenda reiciendis nesciunt numquam similique!"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/1200px-Image_created_with_a_mobile_phone.png"
    />
  </div>
  <div
    className="cardTeacherPhoto"
  >
    <img
      alt="Course 1"
      src="https://dyl80ryjxr1ke.cloudfront.net/external_assets/hero_examples/hair_beach_v1785392215/original.jpeg"
    />
  </div>
  <div
    className="cardContent"
  >
    <div
      className="cardContentTitle"
    >
      Course 1
    </div>
    <div
      className="cardContentDescription"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero non illum est fuga beatae ab, assumenda reiciendis nesciunt numquam similique!
    </div>
  </div>
  <div
    className="cardButtonContainer"
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
      variant="outlined"
    >
      courses:enterButtonLabel
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
